py/test/selenium/webdriver/common/bidi_network_tests.py

```diff
@@ -93,3 +94,24 @@ def test_remove_auth_handler(driver):
     assert callback_id is not None, "Request handler not added"
     driver.network.remove_auth_handler(callback_id)
     assert driver.network.intercepts == [], "Intercept not removed"
+
+
+@pytest.mark.xfail_chrome(reason="Data URLs in Network requests are not implemented in Chrome yet")
+@pytest.mark.xfail_edge(reason="Data URLs in Network requests are not implemented in Edge yet")
+@pytest.mark.xfail_firefox(reason="Data URLs in Network requests are not implemented in Firefox yet")
+def test_handler_with_data_url_request(driver, pages):
+    data_requests = []
+
+    def callback(request: Request):
+        if request.url.startswith("data:"):
+            data_requests.append(request)
+        request.continue_request()
+
+    driver.network.add_request_handler("before_request", callback)
+    url = pages.url("data_url.html")
+    driver.browsing_context.navigate(context=driver.current_window_handle, url=url, wait=ReadinessState.COMPLETE)
+
+    # Assert that the BiDi event was captured.
+    assert len(data_requests) > 0
```

Comment:
This assertion is not correct for what we are trying to test. For verification, try running this test without `xfail`, it will pass.

As per https://github.com/w3c/webdriver-bidi/issues/727#issuecomment-2160160367, `network.beforeRequestSent` will catch the emitted `data:` requests but we can't intercept it. So, `data_requests.append(request)` will successfully append the `data:` request too, which we assert later.

The issue is with `request.continue_request()` which currently is not able to find the `data:` request to continue. We should test `continue_request()` and not `before_request`.

Also, the errors we had in https://github.com/SeleniumHQ/selenium/issues/16279 are silent and don't interrupt the program flow.